src/main/java/Task.java

```diff
@@ -0,0 +1,136 @@
+import java.util.*;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+
+public class Task {
+    public String type;
+    public String name;
+    public String status;
+    public String date;
+    public LocalDate processedDate;
+    public String fileDate;
+    public boolean hasProcessedDate = false;
+
+    public Task(String type, String name) {
+        this.type = type;
+        this.name = name;
+        this.status = "[✗]";
+    }
+
+    public void setDone() {
+        this.status = "[✓]";
+    }
+
+    public int getStatusBinary() {
+        if (this.status == "[✓]") {
+            return 1;
+        } else {
+            return 0;
+        }
+
+    }
+
+    public String getIcon() {
+        if (this.type.equals("todo")) {
+            return "[T]";
+        } else if (this.type.equals("deadline")) {
+            return "[D]";
+        } else {                  // event
+            return "[E]";
+        }
+    }
+
+    /**
+     * Processes the user input date into a localdate format.
+     *
+     * @return String This returns the formatted form of the localdate.
+     */
+
+    public String getProcessedDate() {
+        String dateFormatted = "";
+        this.date.split("\\s");
+        for (int i = 1; i < this.date.split("\\s").length; i++) {
```

Comment:
It might improve efficiency if we save the String[] from the this.date.split("\\s") into a local variable and use that variable for the next few lines.